### PR TITLE
NEW: 'UI vs Game Input' sample.

### PR DESCRIPTION
**Pull request by Rene-Damm**
### Description

Adds a new `UI vs Game Input` sample that goes into the problem of having UI overlaid over a 3D scene and then having to deal with ambiguities between "game input" vs "UI input".

image

### Changes made

- Added a `UIvsGameInput` sample to the package.
- Added a warning to `IsPointerOverGameObject` if called from within action callbacks (or any input processing).
- Added documentation to the UI section of the docs.
- Fixed a minor problem when copypasting bindings in the .inputactions editor.
- Small change such that `OnControlsChanged` is triggered by `PlayerInput` also on the very first time bindings are resolved, not only when controls are changed later. This makes it possible to always respond to control setup changes in the same way.

### Checklist

Before review:

- [x] Changelog entry added.
    - Explains the change in `Changed`, `Fixed`, `Added` sections.
    - For API change contains an example snippet and/or migration example.
    - FogBugz ticket attached, example `([case %number%](https://issuetracker.unity3d.com/issues/...))`.
    - FogBugz is marked as "Resolved" with *next* release version correctly set.
- [x] Tests added/changed, if applicable.
    - Functional tests `Area_CanDoX`, `Area_CanDoX_EvenIfYIsTheCase`, `Area_WhenIDoX_AndYHappens_ThisIsTheResult`.
    - Performance tests.
    - Integration tests.
- [x] Docs for new/changed API's.
    - Xmldoc cross references are set correctly.
    - Added explanation how the API works.
    - Usage code examples added.
    - The manual is updated, if needed.

During merge:

- [ ] Commit message for squash-merge is prefixed with one of the list:
    - `NEW: ___`.
    - `FIX: ___`.
    - `DOCS: ___`.
    - `CHANGE: ___`.
    - `RELEASE: 1.1.0-preview.3`.

### CODE REVIEW

**Comment by ekcoh** [comment]
@Rene-Damm Maybe move this PR into draft status for time being if currently being worked on?

File: Packages/com.unity.inputsystem/Documentation~/UISupport.md [hunk 1]
```diff
@@ -150,18 +151,52 @@ At runtime, the component adds a virtual [Mouse](../api/UnityEngine.InputSystem.
 Note that the resulting [Mouse](../api/UnityEngine.InputSystem.Mouse.html) input is visible in all code that picks up input from the mouse device. You can therefore use the component for mouse simulation elsewhere, not just with [InputSystemUIInputModule](../api/UnityEngine.InputSystem.UI.InputSystemUIInputModule.html).
 
 >[!NOTE]
->Do not set up gamepads and joysticks for [navigation input](#navigation-type-input) while using VirtualMouseInput. If both VirtualMouseInput and navigation are configured, input is triggered twice: once via the pointer input path, and once via the navigation input path. If you encounter problems such as where buttons are pressed twice, this is likely the problem.
+>Do not set up gamepads and joysticks for [navigation input](#navigation-type-input) while using `VirtualMouseInput`. If both `VirtualMouseInput` and navigation are configured, input is triggered twice: once via the pointer input path, and once via the navigation input path. If you encounter problems such as where buttons are pressed twice, this is likely the problem.
 
-# UI Toolkit support
+## UI and game input
```
Comment (ekcoh) [inline Packages/com.unity.inputsystem/Documentation~/UISupport.md:156]: Might this be something we want doc team review of? @Rene-Damm

**Review by ekcoh — APPROVED** [review]
Code looks good to me and sample in its original form works well (tested win + mouse/keyboard). However, in the sample there is an #if false stating that there are two alternative paths available. I tried changing this to #if true to switch path and when I do that a breakpoint in that callback is triggered and fire flag is set but there is no projectile being spawned. Is this an issue or intentional and I do not understand it correctly? Would suspect this might cause confusion regarding the behavior of the sample.

File: Packages/com.unity.inputsystem/Documentation~/UISupport.md (hunk 1 above)
Comment (Rene-Damm) [reply Packages/com.unity.inputsystem/Documentation~/UISupport.md:156]: Yup, this should get a look

**Comment by Rene-Damm** [comment]
> . I tried changing this to #if true to switch path and when I do that a breakpoint in that callback is triggered and fire flag is set but there is no projectile being spawned. Is this an issue or intentional and I do not understand it correctly? Would suspect this might cause confusion regarding the behavior of the sample.

Sounds broken. I'll have a look.

**Pull request merged**